File: 16-range-sum-query(immutable)/answer.py
class NumArray:

    def __init__(self, nums: list[int]):
        self.num_array = nums

        for i in range(1, len(nums)):
            self.num_array[i] += self.num_array[i-1]

    def sumRange(self, left: int, right: int) -> int:
        if left == 0:
            return self.num_array[right]
        else:
            return self.num_array[right] - self.num_array[left - 1]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 16-range-sum-query(immutable)/answer.go
package main

type NumArray []int

func Constructor(nums []int) NumArray {
	var numArray NumArray = nums

	for i := 1; i < len(nums); i++ {
		numArray[i] += numArray[i-1]
	}

	return numArray
}

func (this *NumArray) SumRange(left int, right int) int {
	if left == 0 {
		return (*this)[right]
	}

	return (*this)[right] - (*this)[left-1]
}

/**
 * Your NumArray object will be instantiated and called as such:
 * obj := Constructor(nums);
 * param_1 := obj.SumRange(left,right);
 */
